src/services/document-service.js

```diff
@@ -175,6 +176,12 @@ class DocumentService {
 
         await this.documentRevisionStore.saveDocumentRevision(newRevision, { session });
         await this.documentStore.saveDocument(newDocument, { session });
+
+        if (newDocument.roomId) {
```

Comment:
If the `roomId` is set, we should have also taken the room lock at the beginning of this `try` block, so no other concurrent room updates can take place.

Reply:
Good point, thanks! Took the lock.